### ISSUE TITLE
job-list: stats missing for configured queue

### ISSUE THREAD

**Issue by grondo** [flux-framework/flux-core#5688]
The job-list module only includes stats for queues to which at least one job has been submitted. This means no stats will appear for a configured but unused queue, except in the case where a job is submitted to a queue and moved to another queue.
After a reload, job-list doesn't update stats while processing events from the eventlog, so no stats will appear for an unused queue even in this case.

It would perhaps be better to initialize zeroed queue stats for all configured queues, so even unused queues will appear in `flux job stats` output (which is a reasonable assumption a user of this interface or the corresponding RPC might have)

**Comment by grondo** [comment]
The issue is easily observed by adding a queue and then noting that `flux job stats` contains no queues even though `flux queue list` shows the queue:

```console
$ echo "[queues.batch]" | flux config load
$ flux queue list
QUEUE    DEFAULTTIME  TIMELIMIT     NNODES     NCORES      NGPUS
batch            inf        inf      0-inf      0-inf      0-inf
$ flux job stats | jq .queues
[]
```

**Comment by chu11** [comment]
> It would perhaps be better to initialize zeroed queue stats for all configured queues, so even unused queues will appear in flux job stats output (which is a reasonable assumption a user of this interface or the corresponding RPC might have)

seems like a good idea.  

One possible alternative, if a jobspec or jobspec-update has a queue mentioned, make sure it is job stats.  This could handle the very rare case that a queue existed before, but no longer exists?  ... I'm not sure this is worth the energy of doing though.

**Comment by grondo** [comment]
That doesn't solve the case of a configured queue that has never had a job submitted to it (as in the example above)

**Comment by chu11** [comment]
oh duh, you're right, it would only solve the bug solved by #5685.  So it'd have to be both.
